rudder-web/src/main/webapp/javascript/rudder/angular/quicksearch.js

```diff
@@ -122,54 +122,53 @@ quicksearch.controller('QuicksearchCtrl', function QuicksearchCtrl($scope, $root
       }
     }
   }
-  $scope.getNumResults = function(){
-    var num;
-    var result;
-    var strRegExp = '((\\s*in:\\s*((directive)|(group)|(node)|(parameter)|(rule))+\\s*,*\\s*)|(,\\s*((directive)|(group)|(node)|(parameter)|(rule))+\\s*))';
-    var regexp = new RegExp(strRegExp , "gi");
-    var baseSearch = $('#searchInput').val().replace(regexp, '');
-
-    //Si il n'y a pas de filtre
-    if((($('#searchInput').val().match(regexp))&&(baseSearch.toUpperCase()!==$scope.strBaseSearch.toUpperCase()))||(!$('#searchInput').val().match(regexp))){
-      $scope.strBaseSearch = baseSearch;
-      $scope.filter.all.nbResults = 0;
-      for(filter in $scope.filter){
-        result = $scope.results.hasOwnProperty(filter) ? $scope.results[filter] : false;
-        if(result){
-          num = result[0].originalObject.numbers;
-        }else{
-          num = 0;
+  $scope.getNumResults = function(filter, results){
+	var res=0;
+	if((results)&&(results.length>0)){
+      for(var i=0 ; i<results.length ; i++){
+        if(filter.toUpperCase()=="ALL"){
+          res += results[i].originalObject.header.numbers;
+        }else if(results[i].originalObject.header.type.toUpperCase() == filter.toUpperCase()){
+          res = results[i].originalObject.header.numbers;
         }
-        $scope.filter.all.nbResults += num;
-        $scope.filter[filter].nbResults = num;
       }
-    //Si il y a des filtres
     }
+	return res;
   }
 
   $scope.noSearch = function() {
     return $scope.searchStr.length<1;
   }
-} );
+
+  $scope.checkSearch = function(searching, searchStr){
```

Comment:
Here you could have not passed searchStr and searching because they are defined in scope